apps/backend/src/users/users.service.ts

```diff
@@ -37,6 +38,40 @@ export class UsersService {
     return await this.prisma.user.findMany({ omit: { weight: true } });
   }
 
+  async calBac(authSchId: string): Promise<UserBac> {
+    const user = await this.prisma.user.findUnique({
+      where: { authSchId },
+      select: { weight: true, gender: true },
+    });
+
+    if (!user) {
+      throw new NotFoundException('User not found');
+    }
+
+    const currentTime = new Date();
+    const userWeightInGrams = user.weight * 1000;
+    const genderFactor = user.gender === 'Male' ? 0.68 : 0.55;
```

Comment:
the weight and gender fields are optional on the user model, so we should check if they're there before using them. throw some error if they're not set